specs/core/0_beacon-chain.md

```diff
@@ -787,23 +792,44 @@ Note: We aim to migrate to a S[T/N]ARK-friendly hash function in a future Ethere
 
 `def hash_tree_root(object: SSZSerializable) -> Bytes32` is a function for hashing objects into a single root utilizing a hash tree structure. `hash_tree_root` is defined in the [SimpleSerialize spec](https://github.com/ethereum/eth2.0-specs/blob/master/specs/simple-serialize.md#tree-hash).
 
+#### `slot_to_epoch`
+
+```python
+def slot_to_epoch(slot: SlotNumber) -> EpochNumber:
+    return slot // EPOCH_LENGTH
+```
+
+#### `get_current_epoch`
+
+```python
+def get_current_epoch(state: BeaconState) -> EpochNumber:
+    return slot_to_epoch(state.slot)
+```
+
+#### `get_epoch_start_slot`
+
+```python
+def get_epoch_start_slot(epoch: EpochNumber) -> SlotNumber:
+    return slot_to_epoch(state.slot)
```

Comment:
Ought this be

```python
    return epoch * EPOCH_LENGTH
```

or something like that?

Reply:
It's frustating that `get_block_root` is called only at epoch boundaries, with only one exception.

Reply:
Yup, that's a copy pasta error.

`epoch_start_to_slot` is okay, but I think `epoch_to_slot` is not descriptive enough.